src/cargo-fmt/main.rs

```diff
@@ -246,12 +278,22 @@ fn get_targets(strategy: &CargoFmtStrategy) -> Result<BTreeSet<Target>, io::Erro
     }
 }
 
-fn get_targets_root_only(targets: &mut BTreeSet<Target>) -> Result<(), io::Error> {
-    let metadata = get_cargo_metadata(None, false)?;
-    let current_dir = env::current_dir()?.canonicalize()?;
-    let current_dir_manifest = current_dir.join("Cargo.toml");
+fn get_targets_root_only(
+    manifest_path: Option<&Path>,
+    targets: &mut BTreeSet<Target>,
+) -> Result<(), io::Error> {
+    let metadata = get_cargo_metadata(manifest_path, false)?;
     let workspace_root_path = PathBuf::from(&metadata.workspace_root).canonicalize()?;
-    let in_workspace_root = workspace_root_path == current_dir;
+    let (in_workspace_root, current_dir_manifest) = if manifest_path.is_some() {
+        let target_manifest = manifest_path.unwrap().canonicalize()?;
```

Comment:
If specified, `manifest_path` will already have a `Cargo.toml` suffix so `manifest_path` can just be used directly instead.